docs/develop/func/cookbook.md

```diff
@@ -50,51 +53,60 @@ repeat(count) {
 
 ### How to write a while loop
 
-TODO: please provide some description like in a `Modulo operations` section
+While is useful when we do not know how often to perform a particular action. For example, take a `cell`, which is known to store up to four references to other cells. 
 
 ```func
-int i = 0;
-
-while (i < 50) {
+cell inner_cell = begin_cell() ;; create a new empty builder
+        .store_uint(123, 16) ;; store uint with value 123 and length 16 bits
+        .end_cell(); ;; convert builder to a cell
+
+cell message = begin_cell()
+        .store_ref(inner_cell) ;; store cell as reference
+        .store_ref(inner_cell)
+        .end_cell();
+
+slice msg = message.begin_parse(); ;; convert cell to slice
+while (msg.slice_refs_empty?() != -1) { ;; we should remind that -1 is true
```

Comment:
Is `!= -1` really needed?

Reply:
I wanted to show that in FunC -1 it is true.